File: 2st/Riyeon/짝지어 제거하기.java
import java.util.*;
class Solution{
    public int solution(String s){
        Stack<String> st = new Stack<>();
        for(String str : s.split("")){
            if(!st.isEmpty()){
                if(st.peek().equals(str))
                    st.pop();
                else st.push(str);
            }else st.push(str);
        }
        return st.isEmpty()? 1 : 0;
    }
}

File: 3st/Hyeonseo/com/짝지어 제거하기.js
function solution(s) {
    let stack = [];

    for (let char of s) {
        if (stack.length > 0 && stack[stack.length - 1] === char) {
            stack.pop(); 
        } else {
            stack.push(char); 
        }
    }

    return stack.length === 0 ? 1 : 0;
}

